Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1193,7 +1193,26 @@ class BrowserViewController: UIViewController, SearchBarLocationProvider, Themea
         }
     }
 
-    func showLibrary(panel: LibraryPanelType? = nil) {
+    func showLibrary(panel: LibraryPanelType) {
+        if CoordinatorFlagManager.isLibraryCoordinatorEnabled {
+            DispatchQueue.main.async {
+                self.navigationHandler?.show(homepanelSection: self.toHomepanelSection(panel))
+            }
+        } else {
+            self.showLegacyLibrary(panel: panel)
+        }
+    }
+
+    private func toHomepanelSection(_ panelType: LibraryPanelType) -> Route.HomepanelSection {
+        switch panelType {
+        case .bookmarks: return .bookmarks
+        case .history: return .history
+        case .readingList: return .readingList
+        case .downloads: return .downloads
+        }
+    }
+
+    func showLegacyLibrary(panel: LibraryPanelType? = nil) {
```

Comment:
One thing I just realized we might have todo because of the following comment:
```
// We should not set libraryViewController to nil because the library panel losses the currentState
```

There's a mechanism in place in the current `libraryViewController` to restore the state when presenting. I haven't checked how it works, but we need to ensure we keep that in place. I don't think we want to keep the libraryViewController reference alive like it was done in the past to enable this, so we should restore the state another way IMO. I rename the task https://mozilla-hub.atlassian.net/browse/FXIOS-6607 to keep track of this, so we check this works/enables it with another PR